scripts/tuple_tree_generator/tuple_tree_generator/templates/tuple_tree_gen.py.tpl

```diff
@@ -95,6 +101,20 @@ class 'struct.name'(
 
 ## endfor ##
 
+## for struct in structs ##
+## if struct.abstract -##
+'struct.name'._children = {
+        ##- for child in struct.children -##
+        "'-child.name'": '-child.name' ##- if not loop.last -## , ##- endif -##
+        ##- endfor -##
+    }
+
+## for child in struct.children -##
+force_constructor_kwarg('child.name', "Kind", 'struct.name'Kind.'child.name')
+## endfor ##
```

Comment:
Comment, I've no idea what this does.

Reply:
It's a function that patches the `__init__` of the child class, it does the following:
* Automatically setting the `Kind` field if not specified.
* If `Kind` is present, assert that it is the correct one.
See https://github.com/mrjackv/revng/blob/feature/remove-model-tag/python/revng/model/_util.py#L10

This is the autogenerated version of [this](https://github.com/revng/revng/blob/de033b50f20019a2ca1d996e0b845e3962688cbe/python/revng/model/v1/__init__.py#L17)

Reply:
I mean: add a comment.

Reply:
Ok 👍